File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int p1 = 0, p2 = 0;

        if(s.size() == 0)   return true;

        while(p1 <= s.size()-1){
            if(s[p1] == t[p2]){
                p1++;
                p2++;
            }
            else{
                p2++;
                if(p2>= t.size())   break;
            }
        }
        return p1 == s.size();
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        l1=l2=0
        temp=''
        while l1<len(s) and l2<len(t):
            if s[l1]==t[l2]:
                temp+=s[l1]
                l1+=1
                l2+=1
            elif s[l1]!=t[l2]:
                l2+=1
        print(temp)
        if temp==s:
            return True
        return False